### PR TITLE
Legend offset new attribute

### PR DESCRIPTION
**Pull request by doutriaux1**
@danlipsa a few baselines are still missing (Linux) it is ready for review.

goes with: https://github.com/UV-CDAT/uvcdat-testdata/pull/149

### CODE REVIEW

**Comment by doutriaux1** [comment]
@danlipsa @aashish24 @mcenerney1 this is needed for ACME improvement. Please review.

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 Why do so many baseline change? What is offset? I would squash related commits. A commit such as '￼pep8ed and flake8ed' has no value in history - it only create noise.

**Comment by doutriaux1** [comment]
@danlipsa so many baseline changes because the legend is slightly different now. `offset` is the distance of the legend labels from the colorbar. peped8 and such commits have value to me because they are cosmetic changes and they make it harder to follow the changes that actually happened in the process. Going commit by commit helps reviewing and we can happily skip the pep8 ones. I personally don't like PR with ONE commit, I like to see the work that went in nd follow the thought process. History is mostly useless once a PR is in anyway.

**Comment by danlipsa** [comment]
@doutriaux1 I agree that adding extra commits makes reviewing easier - this is one place many commits is better than a squashed commit. We use the history in ParaView and VTK to track regressions that would otherwise take longer to fix.

**Comment by doutriaux1** [comment]
@danlipsa I see. Feel free to squash commits if it really bothers you.

File: Packages/vcs/vcs/VCS_validation_functions.py [hunk 1]
```diff
@@ -1745,13 +1745,37 @@ def _setpriority(self, value):
...
+
+@arrow.setter
+def arrow(self, value):
+    self._arrow = checkNumber(self, "arrow", value, 0., None)
+
+
+@property
+def offset(self):
+    """offset position in % of page (can be negative)"""
+    return self._offset
```
Comment (danlipsa) [inline Packages/vcs/vcs/VCS_validation_functions.py]: @doutriaux1 Can you expand the comment for offset to include the detailed explanation in the comments section?

File: Packages/vcs/vcs/Plegend.py [hunk 2]
```diff
@@ -115,6 +120,8 @@ def __init__(self, member):
         self.y1 = 0.129999995232
         self.x2 = 0.949999988079
         self.y2 = 0.159999996424
+        self.offset = .015
```
Comment (danlipsa) [inline Packages/vcs/vcs/Plegend.py]: @doutriaux1 Would it be possible to have a smaller value here so that most baselines do not change. It seems that for the first images at least, the change is the bigger offset?

**Comment by danlipsa** [comment]
@doutriaux1 Besides my comments, the changes look reasonable.

**Comment by danlipsa** [comment]
@doutriaux1 I can live without squashed commits :-) I thought it might help us if we need to go back to a previous valid version. But maybe this is more than we need for uvcdat.

**Comment by doutriaux1** [comment]
@danlipsa this is ready for final review all test pass on my mac and linux, and they should all pass on travis once this is merged in fix_travis.I can merge in master.

File: CMake/cdat_modules_extra/runtest.in [hunk 3]
```diff
@@ -1,6 +1,9 @@
 #!/bin/bash
 echo "ACTIVATING ENV:"@CONDA_ENVIRONMENT_NAME@
-source activate @CONDA_ENVIRONMENT_NAME@
+while [[ `which python` != *"@CONDA_ENVIRONMENT_NAME@"* ]]
+do
+    source activate @CONDA_ENVIRONMENT_NAME@
+done
```
Comment (danlipsa) [inline CMake/cdat_modules_extra/runtest.in]: @doutriaux1 Why do you need to do this?

**Comment by danlipsa** [comment]
LGTM :+1:

**Comment by danlipsa** [comment]
@doutriaux1 LGTM :+1:

File: CMake/cdat_modules_extra/runtest.in (hunk 3 above)
Comment (doutriaux1) [inline CMake/cdat_modules_extra/runtest.in]: @danlipsa because sometimes the environement does not get activated. I suspect some concurency issues within conda, too many process trying to activate at same time, so I keep trying until it's actually activated. We could/should probably improve the test to see if it's activated (using `conda env list` or something like that) but that would be another PR.

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 Here is some discussion on squshing commits
http://programmers.stackexchange.com/questions/263164/why-squash-git-commits-for-pull-requests
No pressure though :smile:

**Comment by doutriaux1** [comment]
not convinced. If it's for the reviewer that cares for the "net effect" then just look at this on the PR. Now if that is not enough and the reviewer wants to look at the history in detail then it's easy better I think to ignore the `pep8ed` commit and look at the previous one which has only the changes made, rather than a bi commit with a bunch of useless `pep8ed` changes in it that had to what really changed. Anyhow, feel free to squash them if you feel like the history would be better.

File: CMake/cdat_modules_extra/runtest.in (hunk 3 above)
Comment (danlipsa) [inline CMake/cdat_modules_extra/runtest.in]: @doutriaux1 I see. Thanks!
